internal/transport/http2_client.go

```diff
@@ -824,6 +825,12 @@ func (t *http2Client) NewStream(ctx context.Context, callHdr *CallHdr) (*Stream,
 			return nil, &NewStreamError{Err: err, AllowTransparentRetry: true}
 		}
 		if success {
+			// drain client transport if nextID > MaxStreamID, which is set to
+			// 75% of math.MaxUint32, which then signals gRPC to restart transport
+			// for subsequent RPCs.
+			if t.nextID > MaxStreamIDForTesting {
+				transportDrainRequired = true
+			}
```

Comment:
Oh whoops.  You have a race.  This bool needs to be set inside `checkForStreamQuota` then read here (or below as is and ignore here).

Reply:
im made the change now